shared/macros-bash.jinja

```diff
@@ -319,6 +319,57 @@ else
 fi
 {{%- endmacro -%}}
 
+{{#
+# Sets PAM faillock module options and values.
+# Also it adds pam_faillock.so as required module for account.
+# option: faillock option eg. deny, unlock_time
+# value: value of option
+#}}
+{{%- macro bash_set_faillock_option(option, value) -%}}
+AUTH_FILES[0]="/etc/pam.d/system-auth"
+AUTH_FILES[1]="/etc/pam.d/password-auth"
+
+for pam_file in "${AUTH_FILES[@]}"
+do
+  # is auth required pam_faillock.so preauth present?
+  if grep -qE "^\s*auth\s+required\s+pam_faillock\.so\s+preauth.*$" "$pam_file" ; then
+    # is the option set?
+    if grep -qE "^\s*auth\s+required\s+pam_faillock\.so\s+preauth.*{{{ option }}}=([0-9]*).*$" "$pam_file" ; then
+      # just change the value of option to a correct value
+      sed -i --follow-symlinks "s/\(^auth.*required.*pam_faillock.so.*preauth.*silent.*\)\({{{ option }}} *= *\).*/\1\2{{{ value }}}/" "$pam_file"
```

Comment:
Please use single quotes whenever you can, as inside double quotes, the backslash should be escaped twice, which may be confusing.